lib/metrics_collector.rb

```diff
@@ -80,13 +76,12 @@ def run(inventoried_time, morefs_to_meter = [])
       end
       @readings << reading
     end
-
     logger.info "Adding #{@readings.size} readings to metrics collection"
 
     # We don't update the status until all metrics have been collected. This way, if
     #  saving fails, the inventory will remain queued for metering (i.e., it will get
     #  picked up again for processing automatically)
-
+    # Apply Lock over here??????
```

Comment:
Maybe it could be great if you include a "TODO" or "FIXME" at the beginning of this comment.